service/storage_service.cc

```diff
@@ -1105,6 +1105,14 @@ class join_node_rpc_handshaker : public service::group0_handshaker {
 };
 
 future<> storage_service::raft_initialize_discovery_leader(raft::server& raft_server, const join_node_request_params& params) {
+    // No other server can become a member of the cluster until
+    // we finish adding ourselves. This is ensured by
+    // waiting in join_node_request_handler for
+    // _topology_state_machine._topology.normal_nodes to be not empty.
+    // We cannot mark ourselves as dead during this process.
+    // Since there's no chance of raft operations
+    // getting stuck, we don't use raft_timeout{} here.
```

Comment:
We could use it like a sanity check.

Reply:
I'm not sure timeout is a good sanity check here. The function `raft_initialize_discovery_leader` is part of the node startup routine, which already includes several valid uses of `raft_timeout{}`. If a timeout occurs here, it would be challenging to differentiate it from other timeouts. Not having a timeout might actually serve as a more effective sanity check - we can't overlook the bootstrapping process getting stuck.

Or maybe we should use `raft_timeout{}`, try/catch our timeout exception and transform it to `on_internal_error` here.

Reply:
> Or maybe we should use raft_timeout{}, try/catch our timeout exception and transform it to on_internal_error here.

Good idea, but we'd have to use something more specific than `runtime_error`

Reply:
yes